cpp/src/arrow/flight/sql/example/sqlite_type_info.cc

```diff
@@ -0,0 +1,116 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/flight/sql/example/sqlite_type_info.h"
+
+#include "arrow/flight/sql/server.h"
+#include "arrow/flight/sql/types.h"
+#include "arrow/record_batch.h"
+#include "arrow/testing/gtest_util.h"
+
+namespace arrow {
+namespace flight {
+namespace sql {
+namespace example {
+
+std::shared_ptr<RecordBatch> DoGetTypeInfoResult() {
+  auto type_name_array =
+      ArrayFromJSON(utf8(), R"(["bit", "tinyint", "bigint", "longvarbinary",
+                            "varbinary", "text", "longvarchar", "char",
+                            "integer", "smallint", "float", "double",
+                            "numeric", "varchar", "date", "time",
+                            "timestamp"])");
+  auto data_type = ArrayFromJSON(
+      int32(), R"([-7, -6, -5, -4, -3, -1, -1, 1, 4, 5, 6, 8, 8, 12, 91, 92, 93])");
+  auto column_size = ArrayFromJSON(
+      int32(),
+      R"([1, 3, 19, 65536, 255, 65536, 65536, 255, 9, 5, 7, 15, 15, 255, 10, 8, 32])");
+  auto literal_prefix = ArrayFromJSON(
+      utf8(),
+      R"([null, null, null, null, null, "'", "'", "'", null, null, null, null, null, "'", "'", "'", "'"])");
+  auto literal_suffix = ArrayFromJSON(
+      utf8(),
+      R"([null, null, null, null, null, "'", "'", "'", null, null, null, null, null, "'", "'", "'", "'"])");
+  auto create_params = ArrayFromJSON(
+      list(field("item", utf8(), false)),
+      R"([[], [], [], [], [], ["length"], ["length"], ["length"], [], [], [], [], [], ["length"], [], [], []])");
+  auto nullable =
+      ArrayFromJSON(int32(), R"([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1])");
+  auto case_sensitive =
+      ArrayFromJSON(boolean(), R"([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])");
+  auto serachable =
+      ArrayFromJSON(int32(), R"([3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3])");
+  auto unsigned_attribute =
+      ArrayFromJSON(boolean(), R"([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])");
+  auto fixed_prec_scale =
+      ArrayFromJSON(boolean(), R"([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])");
+  auto auto_unique_value =
+      ArrayFromJSON(boolean(), R"([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])");
+  auto local_type_name =
+      ArrayFromJSON(utf8(), R"(["bit", "tinyint", "bigint", "longvarbinary",
+                          "varbinary", "text", "longvarchar", "char",
+                          "integer", "smallint", "float", "double",
+                          "numeric", "varchar", "date", "time",
+                          "timestamp"])");
+  auto minimal_scale =
+      ArrayFromJSON(int32(), R"([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])");
+  auto maximum_scale =
+      ArrayFromJSON(int32(), R"([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])");
+  auto sql_data_type =
```

Comment:
Can just reuse the `data_type` array in the call to `RecordBatch::Make` below.

Reply:
Fixed